### ISSUE TITLE
Samples restrictive

### ISSUE THREAD

**Issue by Yahweasel** [w3c/webcodecs#406]
The video/audio samples provided in `samples/` all have issues that make them difficult to use to demonstrate WebCodecs _in isolation_ (which, for me, makes it difficult to use them with a polyfill or, thus, to demonstrate or test said polyfill):

- MP4 Decoding uses a codec controlled by the Misanthropic Patent Extortion Gang, so it's unlikely to be supported by any fully F/OSS implementations.
- Webcam in Worker uses the MediaStreamTrackProcessor API, which is a separate API entirely (and also experimental, and somewhere between difficult and impossible to polyfill).
- Capture to File uses MediaStreamTrackProcessor _and_ the FileSystem API, which is _also_ a separate, experimental API.

Admittedly, this is of particular nuisance only to me, because it means I can't say "supports the WebCodecs samples", but regardless it's a breadth issue.

It'd be nice to see simple transcoding (VP9/FLAC to VP8/Opus, for instance), or even just a trivial decoder or encoder with no surrounding bells and whistles. I have some in-progress samples (using my polyfill) at https://github.com/ennuicastr/libavjs-webcodecs-polyfill/tree/master/samples/ .

**Comment by Yahweasel** [comment]
I'm going to necro this thread: I have tons of samples from both the polyfill and a new bridge I wrote that links between any file format supported by ffmpeg/libav and WebCodecs. I don't know how all the policies work, but would it be possible/reasonable for me to PR samples?

**Comment by padenot** [comment]
Yeah, the idea being that it more or less demonstrates something you'd do, e.g. encoding the webcam, or transcoding a media file, etc., simple but real uses.

We should also add opus in the mp4 example, and it'll be fine on that front, av1/vp8/vp9/opus. 

@tidoust, how would that work in terms of IPR?

**Comment by Yahweasel** [comment]
Just for reference, my samples in the polyfill (https://github.com/ennuicastr/libavjs-webcodecs-polyfill/tree/master/samples) are:
* Decoding an audio file (FLAC, Opus)
* Encoding an audio file (FLAC, Opus)
* Decoding an AV file (VP8, Opus)
* Encoding a video file (VP8)
* Encoding the webcam to a video file (VP8)
* Encoding and decoding the webcam "on the fly" (just to demonstrate both sides)

Of course, all of the codec-specific formats are easily funged into any other codec. I use Matroska/WebM in all cases, but libav supports Misanthropic Patent Extortion Gang 4 just fine.

My main sample in the bridge (https://github.com/Yahweasel/libavjs-webcodecs-bridge/tree/main/samples/transcoder) is an any-to-any transcoder.

**Comment by tidoust** [comment]
> @tidoust, how would that work in terms of IPR?

Anything you worry about in particular?

Samples are not published with the spec, the Patent Policy does not apply to them. Even if they were, content would be informative. Contributing samples to the repository is like contributing to any regular open source project, noting the license used is the [W3C Software and Document License](https://www.w3.org/copyright/software-license-2023/) (which is close to an MIT license).

We may still have to tell the IPR bot that a PR that touches samples is non-substantive on a PR-by-PR basis, but that should remain manageable.

(This makes me realize that the [`CONTRIBUTING.md`](https://github.com/w3c/webcodecs/blob/main/CONTRIBUTING.md) file in this repo (and probably other repos as well), is likely the result of a copy-and-paste of an older version of the file. The [latest template](https://github.com/w3c/ash-nazg/blob/master/templates/WG-CONTRIBUTING-SW.md) is more explicit that the instructions apply to **substantive** contributions. I'll look into refreshing these files.)

**Comment by padenot** [comment]
Yeah, all good then, I wasn't sure, so I figured I'd ask.

@yeahweasel, transcoding, encoding a canvas to video, encoding PCM to audio (possible both at the same time), encoding the webcam seem like good candidates, in the sense that I'm pretty sure I've wrote samples for people in the past about those.
